daemon/algod/api/server/v2/handlers.go

```diff
@@ -781,6 +787,24 @@ func (v2 *Handlers) GetStatus(ctx echo.Context) error {
 		CatchpointAcquiredBlocks:    &stat.CatchpointCatchupAcquiredBlocks,
 	}
 
+	upgradePropose := string(latestBlkHdr.UpgradeVote.UpgradePropose)
+	if upgradePropose != "" && latestBlkHdr.NextProtocolVoteBefore > stat.LastRound {
+		votesToGo := uint64(latestBlkHdr.NextProtocolVoteBefore) - uint64(stat.LastRound)
+		consensus := config.Consensus[protocol.ConsensusCurrentVersion]
+		upgradeVoteRounds := consensus.UpgradeVoteRounds
+		upgradeThreshold := consensus.UpgradeThreshold
+		votes := uint64(upgradeVoteRounds) - votesToGo
+		votesYes := latestBlkHdr.UpgradeState.NextProtocolApprovals
+		votesNo := votes - votesYes
+		upgradeDelay := uint64(latestBlkHdr.UpgradeVote.UpgradeDelay)
+
+		response.UpgradePropose = &upgradePropose
+		response.UpgradeThreshold = &upgradeThreshold
+		response.UpgradeApprove = &latestBlkHdr.UpgradeApprove
+		response.UpgradeDelay = &upgradeDelay
+		response.UpgradeNo = &votesNo
+	}
```

Comment:
Discussed offline, but I think it's useful to write down here for posterity...

I think it's important to put this logic in the [`node.Status`](https://github.com/algorand/go-algorand/blob/master/node/node.go#L655) function.  
Reason being is that the `Status` call uses the `ledger.BlockHdr` call used above internally to fill in the output. So the following sequence of events becomes problematic.
1. Node.Status is called returning status info for the latest round, round X.
2. Either agreement or catchup advances the latest round in the ledger to X+1.
3. Ledger.BlockHdr is called for the latest round, X+1.  

Now the data in `latestBlkHdr` potentially conflicts the data in `stat`.

Reply:
done